src/main/java/net/dv8tion/jda/client/entities/Application.java

```diff
@@ -0,0 +1,295 @@
+
+import java.util.Collection;
+import java.util.List;
+import net.dv8tion.jda.client.managers.ApplicationManager;
+import net.dv8tion.jda.client.managers.ApplicationManagerUpdatable;
+import net.dv8tion.jda.core.JDA;
+import net.dv8tion.jda.core.Permission;
+import net.dv8tion.jda.core.entities.ISnowflake;
+import net.dv8tion.jda.core.managers.ChannelManager;
+import net.dv8tion.jda.core.requests.RestAction;
+
+/** 
+ * Represents a Discord Application from it's owning client point of view
+ * 
+ * @since  JDA 3.0
+ * @author Aljoscha Grebe
+ * 
+ * @see <a href="https://discordapp.com/developers/applications/me">Discord Documentation - My Apps</a>
+ * @see {@link net.dv8tion.jda.client.JDAClient#getApplications() JDAClient#getApplications()}
+ * @see {@link net.dv8tion.jda.client.JDAClient#getApplicationById(String) JDAClient#getApplicationById(String)}
+ */
+public interface Application extends ISnowflake
+{
+
+    /**
+     * Creates a new Bot for this Application.
+     * <b>This cannot be undone!</b>
+     * A new Bot will only be created if no bot-account is already assigned, otherwise the existing one is returned.
+     * A newly created Bot-account will have its name set to the name of the Application.
+     *
+     * @return {@link net.dv8tion.jda.core.requests.RestAction RestAction} -
+     *         Type: {@link Application.Bot}
+     *         <br>The created bot account of this application. 
+     */
+    RestAction<Application.Bot> createBot();
+
+    /**
+     * Deletes this Application and its assigned Bot (if present).
+     * <b>This cannot be undone!</b>
+     *
+     * @return {@link net.dv8tion.jda.core.requests.RestAction RestAction} -
+     *         Type: {@link Void}
+     *         <br>The RestAction to delete this Application.
+     */
+    RestAction<Void> delete();
+
+    /**
+     * Whether the bot requires code grant to invite or not. 
+     * If your application requires multiple scopes then you may need the full OAuth2 flow to ensure a
+     * bot doesn't join before your application is granted a token.
+     * 
+     * @return whether the bot requires code grant
+     */
+    boolean doesBotRequireCodeGrant();
+
+    /**
+     * Returns the Bot assigned to this Application
+     * 
+     * @return The {@link Application.Bot} assigned to this application, or null if no bot is assigned
+     */
+    Application.Bot getBot();
+
+    /**
+     * Returns the description of the application.
+     * 
+     * @return The description of the application or an empty {@link String} if no description is defined.
+     */
+    String getDescription();
+
+    /**
+     * Returns the flags for this application. It's used for whitelisted apps.
+     * 
+     * @return The application flags.
+     */
+    int getFlags();
+
+    /**
+     * Returns the icon id of the application.
+     * 
+     * @return The iconId of the application or null if no icon is defined.
+     */
+    String getIconId();
```

Comment:
this isn't about guilds though, the application page simply just display the background color